### PR TITLE
Upgrade to Realm Core v13.22.0

### PR DESCRIPTION
**Pull request by kneth**
## What, How & Why?

This closes https://github.com/realm/realm-js/issues/6158

## ☑️ ToDos

* [x] 📝 Changelog entry
* [ ] 📝 `Compatibility` label is updated or copied from previous entry
* [ ] 📝 Update `COMPATIBILITY.md`
* [ ] 🚦 Tests
* [ ] 📦 Updated internal package version in consuming `package.json`s (if updating internal packages)
* [ ] 📱 Check the React Native/other sample apps work if necessary
* [ ] 💥 `Breaking` label has been applied or is not necessary

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,18 +7,24 @@
 * None
 
 ### Fixed
-* <How to hit and notice issue? what was the impact?> ([#????](https://github.com/realm/realm-js/issues/????), since v?.?.?)
-* None
+* Receiving a `write_not_allowed` error from the server would have led to a crash. ([realm/realm-core#6978](https://github.com/realm/realm-core/issues/6978), since v11.5.0)
+* If querying over a geospatial dataset that had some objects with a type property set to something other than `Point` (case insensitive) an exception would have been thrown. Instead of disrupting the query, those objects are now just ignored. ([realm/realm-core#6989](https://github.com/realm/realm-core/issues/6989), since v12.0.0)
+* If a user was logged out while an access token refresh was in progress, the refresh completing would mark the user as logged in again and the user would be in an inconsistent state. ([realm/realm-core#6837](https://github.com/realm/realm-core/pull/6837), since v10.0.0)
+* Logging into a single user using multiple auth providers created a separate `Realm.User` per auth provider. This mostly worked, but had some quirks:
```
Comment (elle-j) [inline CHANGELOG.md:13]:
```suggestion
* Logging in a single user using multiple auth providers created a separate `Realm.User` per auth provider. This mostly worked, but had some quirks:
```

File: packages/realm/src/app-services/User.ts [hunk 2]
```diff
@@ -135,13 +135,18 @@ export class User<
...
+      const type = this.internal.identities[0].providerType;
+      if (isProviderType(type)) {
+        return type;
+      } else {
+        throw new Error(`Unexpected provider type: ${type}`);
+      }
     } else {
-      throw new Error(`Unexpected provider type: ${type}`);
+      throw new Error("No provider found");
     }
```
Comment (elle-j) [inline packages/realm/src/app-services/User.ts:150]: One idea here is to access `this.internal.identities` only 1 time. Perhaps something like this?
```ts
get providerType(): ProviderType {
  const [identity] = this.internal.identities;
  if (!identity) {
    throw new Error("No provider found");
  }
  const type = identity.providerType;
  if (isProviderType(type)) {
    return type;
  }
  throw new Error(`Unexpected provider type: ${type}`);
}
```

**Review by elle-j — APPROVED** [review]
🙂

File: packages/realm/src/app-services/User.ts (hunk 2 above)
Comment (kneth) [reply packages/realm/src/app-services/User.ts]: I like it :-)

**Pull request merged**